test/global-setup.spec.ts

```diff
@@ -88,7 +88,30 @@ test('globalTeardown does not run when globalSetup times out', async ({ runInlin
       });
     `,
   });
-  expect(result.skipped).toBe(1);
+  // We did not collect tests, so everything should be zero.
+  expect(result.skipped).toBe(0);
+  expect(result.passed).toBe(0);
+  expect(result.failed).toBe(0);
   expect(result.output).toContain('Timed out waiting 1s for the entire test run');
   expect(result.output).not.toContain('teardown=');
 });
+
+test('globalSetup should be run before requiring tests', async ({ runInlineTest }) => {
+  const { passed } = await runInlineTest({
+    'folio.config.ts': `
+      folio.globalSetup(async () => {
```

Comment:
This test doesn't really explore the dynamics of globalSetup. Is it run once per test file? Once total? How is it different from just running code top level in the config? I expected the test to answer these questions for me.

Reply:
There are more tests in this file, would you like to take a look at them?